internal/query/projection/user.go

```diff
@@ -296,6 +298,34 @@ func (p *userProjection) Reducers() []handler.AggregateReducer {
 					Event:  user.MachineSecretRemovedType,
 					Reduce: p.reduceMachineSecretRemoved,
 				},
+				{
```

Comment:
i noticed that we currently also unset it in case of a phone removal: https://github.com/zitadel/zitadel/blob/62804ca45f804322e54831aeb2c9be9128ae60ca/internal/user/repository/view/model/user.go#L303-L308
i think the idea was generally to unset it if any event occurs related to 2FA (either verified or removed). The login will check if any 2FA is set up, if not will check on the skip time.
So in case of any unset, they will be prompted the next time they log in, no matter the last actual skip and duration set in the policy.